// @swisseph/core/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  Asteroid: () => Asteroid,
  AsteroidOffset: () => AsteroidOffset,
  CalculationFlag: () => CalculationFlag,
  CalculationFlags: () => CalculationFlags,
  CalendarType: () => CalendarType,
  CommonCalculationFlags: () => CommonCalculationFlags,
  CommonEclipseTypes: () => CommonEclipseTypes,
  DateTimeImpl: () => DateTimeImpl,
  EclipseType: () => EclipseType,
  EclipseTypeFlags: () => EclipseTypeFlags,
  FictitiousPlanet: () => FictitiousPlanet,
  HousePoint: () => HousePoint,
  HouseSystem: () => HouseSystem,
  LunarEclipseImpl: () => LunarEclipseImpl,
  LunarPoint: () => LunarPoint,
  NumberOfPlanets: () => NumberOfPlanets,
  Planet: () => Planet,
  PlanetaryMoonOffset: () => PlanetaryMoonOffset,
  RiseTransitFlag: () => RiseTransitFlag,
  SiderealMode: () => SiderealMode,
  SolarEclipseImpl: () => SolarEclipseImpl,
  normalizeEclipseTypes: () => normalizeEclipseTypes,
  normalizeFlags: () => normalizeFlags
});
module.exports = __toCommonJS(index_exports);

// src/enums.ts
var CalendarType = /* @__PURE__ */ ((CalendarType2) => {
  CalendarType2[CalendarType2["Julian"] = 0] = "Julian";
  CalendarType2[CalendarType2["Gregorian"] = 1] = "Gregorian";
  return CalendarType2;
})(CalendarType || {});
var Planet = /* @__PURE__ */ ((Planet2) => {
  Planet2[Planet2["Sun"] = 0] = "Sun";
  Planet2[Planet2["Moon"] = 1] = "Moon";
  Planet2[Planet2["Mercury"] = 2] = "Mercury";
  Planet2[Planet2["Venus"] = 3] = "Venus";
  Planet2[Planet2["Mars"] = 4] = "Mars";
  Planet2[Planet2["Jupiter"] = 5] = "Jupiter";
  Planet2[Planet2["Saturn"] = 6] = "Saturn";
  Planet2[Planet2["Uranus"] = 7] = "Uranus";
  Planet2[Planet2["Neptune"] = 8] = "Neptune";
  Planet2[Planet2["Pluto"] = 9] = "Pluto";
  Planet2[Planet2["Earth"] = 14] = "Earth";
  Planet2[Planet2["EclipticNutation"] = -1] = "EclipticNutation";
  Planet2[Planet2["FixedStar"] = -10] = "FixedStar";
  return Planet2;
})(Planet || {});
var LunarPoint = /* @__PURE__ */ ((LunarPoint2) => {
  LunarPoint2[LunarPoint2["MeanNode"] = 10] = "MeanNode";
  LunarPoint2[LunarPoint2["TrueNode"] = 11] = "TrueNode";
  LunarPoint2[LunarPoint2["MeanApogee"] = 12] = "MeanApogee";
  LunarPoint2[LunarPoint2["OsculatingApogee"] = 13] = "OsculatingApogee";
  LunarPoint2[LunarPoint2["InterpolatedApogee"] = 21] = "InterpolatedApogee";
  LunarPoint2[LunarPoint2["InterpolatedPerigee"] = 22] = "InterpolatedPerigee";
  return LunarPoint2;
})(LunarPoint || {});
var Asteroid = /* @__PURE__ */ ((Asteroid2) => {
  Asteroid2[Asteroid2["Chiron"] = 15] = "Chiron";
  Asteroid2[Asteroid2["Pholus"] = 16] = "Pholus";
  Asteroid2[Asteroid2["Ceres"] = 17] = "Ceres";
  Asteroid2[Asteroid2["Pallas"] = 18] = "Pallas";
  Asteroid2[Asteroid2["Juno"] = 19] = "Juno";
  Asteroid2[Asteroid2["Vesta"] = 20] = "Vesta";
  return Asteroid2;
})(Asteroid || {});
var FictitiousPlanet = /* @__PURE__ */ ((FictitiousPlanet2) => {
  FictitiousPlanet2[FictitiousPlanet2["Cupido"] = 40] = "Cupido";
  FictitiousPlanet2[FictitiousPlanet2["Hades"] = 41] = "Hades";
  FictitiousPlanet2[FictitiousPlanet2["Zeus"] = 42] = "Zeus";
  FictitiousPlanet2[FictitiousPlanet2["Kronos"] = 43] = "Kronos";
  FictitiousPlanet2[FictitiousPlanet2["Apollon"] = 44] = "Apollon";
  FictitiousPlanet2[FictitiousPlanet2["Admetos"] = 45] = "Admetos";
  FictitiousPlanet2[FictitiousPlanet2["Vulkanus"] = 46] = "Vulkanus";
  FictitiousPlanet2[FictitiousPlanet2["Poseidon"] = 47] = "Poseidon";
  FictitiousPlanet2[FictitiousPlanet2["Isis"] = 48] = "Isis";
  FictitiousPlanet2[FictitiousPlanet2["Nibiru"] = 49] = "Nibiru";
  FictitiousPlanet2[FictitiousPlanet2["Harrington"] = 50] = "Harrington";
  FictitiousPlanet2[FictitiousPlanet2["NeptuneLeverrier"] = 51] = "NeptuneLeverrier";
  FictitiousPlanet2[FictitiousPlanet2["NeptuneAdams"] = 52] = "NeptuneAdams";
  FictitiousPlanet2[FictitiousPlanet2["PlutoLowell"] = 53] = "PlutoLowell";
  FictitiousPlanet2[FictitiousPlanet2["PlutoPickering"] = 54] = "PlutoPickering";
  FictitiousPlanet2[FictitiousPlanet2["Vulcan"] = 55] = "Vulcan";
  FictitiousPlanet2[FictitiousPlanet2["WhiteMoon"] = 56] = "WhiteMoon";
  FictitiousPlanet2[FictitiousPlanet2["Proserpina"] = 57] = "Proserpina";
  FictitiousPlanet2[FictitiousPlanet2["Waldemath"] = 58] = "Waldemath";
  return FictitiousPlanet2;
})(FictitiousPlanet || {});
var HouseSystem = /* @__PURE__ */ ((HouseSystem2) => {
  HouseSystem2["Placidus"] = "P";
  HouseSystem2["Koch"] = "K";
  HouseSystem2["Porphyrius"] = "O";
  HouseSystem2["Regiomontanus"] = "R";
  HouseSystem2["Campanus"] = "C";
  HouseSystem2["Equal"] = "A";
  HouseSystem2["VehlowEqual"] = "V";
  HouseSystem2["WholeSign"] = "W";
  HouseSystem2["Meridian"] = "X";
  HouseSystem2["Azimuthal"] = "H";
  HouseSystem2["PolichPage"] = "T";
  HouseSystem2["Alcabitus"] = "B";
  HouseSystem2["Morinus"] = "M";
  return HouseSystem2;
})(HouseSystem || {});
var HousePoint = /* @__PURE__ */ ((HousePoint2) => {
  HousePoint2[HousePoint2["Ascendant"] = 0] = "Ascendant";
  HousePoint2[HousePoint2["MC"] = 1] = "MC";
  HousePoint2[HousePoint2["ARMC"] = 2] = "ARMC";
  HousePoint2[HousePoint2["Vertex"] = 3] = "Vertex";
  HousePoint2[HousePoint2["EquatorialAscendant"] = 4] = "EquatorialAscendant";
  HousePoint2[HousePoint2["CoAscendant1"] = 5] = "CoAscendant1";
  HousePoint2[HousePoint2["CoAscendant2"] = 6] = "CoAscendant2";
  HousePoint2[HousePoint2["PolarAscendant"] = 7] = "PolarAscendant";
  return HousePoint2;
})(HousePoint || {});
var CalculationFlag = /* @__PURE__ */ ((CalculationFlag2) => {
  CalculationFlag2[CalculationFlag2["JPLEphemeris"] = 1] = "JPLEphemeris";
  CalculationFlag2[CalculationFlag2["SwissEphemeris"] = 2] = "SwissEphemeris";
  CalculationFlag2[CalculationFlag2["MoshierEphemeris"] = 4] = "MoshierEphemeris";
  CalculationFlag2[CalculationFlag2["Heliocentric"] = 8] = "Heliocentric";
  CalculationFlag2[CalculationFlag2["TruePositions"] = 16] = "TruePositions";
  CalculationFlag2[CalculationFlag2["J2000"] = 32] = "J2000";
  CalculationFlag2[CalculationFlag2["NoNutation"] = 64] = "NoNutation";
  CalculationFlag2[CalculationFlag2["Speed3"] = 128] = "Speed3";
  CalculationFlag2[CalculationFlag2["Speed"] = 256] = "Speed";
  CalculationFlag2[CalculationFlag2["NoGravitationalDeflection"] = 512] = "NoGravitationalDeflection";
  CalculationFlag2[CalculationFlag2["NoAberration"] = 1024] = "NoAberration";
  CalculationFlag2[CalculationFlag2["Equatorial"] = 2048] = "Equatorial";
  CalculationFlag2[CalculationFlag2["XYZ"] = 4096] = "XYZ";
  CalculationFlag2[CalculationFlag2["Radians"] = 8192] = "Radians";
  CalculationFlag2[CalculationFlag2["Barycentric"] = 16384] = "Barycentric";
  CalculationFlag2[CalculationFlag2["Topocentric"] = 32768] = "Topocentric";
  CalculationFlag2[CalculationFlag2["Sidereal"] = 65536] = "Sidereal";
  CalculationFlag2[CalculationFlag2["ICRS"] = 131072] = "ICRS";
  CalculationFlag2[CalculationFlag2["DpsidepsIAU1980"] = 262144] = "DpsidepsIAU1980";
  CalculationFlag2[CalculationFlag2["JPLHorizons"] = 524288] = "JPLHorizons";
  CalculationFlag2[CalculationFlag2["JPLHorizonsApprox"] = 1048576] = "JPLHorizonsApprox";
  return CalculationFlag2;
})(CalculationFlag || {});
var CommonCalculationFlags = {
  /** Astrometric positions (no aberration or gravitational deflection) */
  Astrometric: 1024 /* NoAberration */ | 512 /* NoGravitationalDeflection */,
  /** Default flags for Swiss Ephemeris with speed */
  DefaultSwissEphemeris: 2 /* SwissEphemeris */ | 256 /* Speed */,
  /** Default flags for Moshier with speed */
  DefaultMoshier: 4 /* MoshierEphemeris */ | 256 /* Speed */
};
var EclipseType = /* @__PURE__ */ ((EclipseType2) => {
  EclipseType2[EclipseType2["Central"] = 1] = "Central";
  EclipseType2[EclipseType2["NonCentral"] = 2] = "NonCentral";
  EclipseType2[EclipseType2["Total"] = 4] = "Total";
  EclipseType2[EclipseType2["Annular"] = 8] = "Annular";
  EclipseType2[EclipseType2["Partial"] = 16] = "Partial";
  EclipseType2[EclipseType2["AnnularTotal"] = 32] = "AnnularTotal";
  EclipseType2[EclipseType2["Penumbral"] = 64] = "Penumbral";
  return EclipseType2;
})(EclipseType || {});
var CommonEclipseTypes = {
  /** All types of solar eclipses */
  AllSolar: 1 /* Central */ | 2 /* NonCentral */ | 4 /* Total */ | 8 /* Annular */ | 16 /* Partial */ | 32 /* AnnularTotal */,
  /** All types of lunar eclipses */
  AllLunar: 4 /* Total */ | 16 /* Partial */ | 64 /* Penumbral */
};
var SiderealMode = /* @__PURE__ */ ((SiderealMode2) => {
  SiderealMode2[SiderealMode2["FaganBradley"] = 0] = "FaganBradley";
  SiderealMode2[SiderealMode2["Lahiri"] = 1] = "Lahiri";
  SiderealMode2[SiderealMode2["DeLuce"] = 2] = "DeLuce";
  SiderealMode2[SiderealMode2["Raman"] = 3] = "Raman";
  SiderealMode2[SiderealMode2["Ushashashi"] = 4] = "Ushashashi";
  SiderealMode2[SiderealMode2["Krishnamurti"] = 5] = "Krishnamurti";
  SiderealMode2[SiderealMode2["DjwhalKhul"] = 6] = "DjwhalKhul";
  SiderealMode2[SiderealMode2["Yukteshwar"] = 7] = "Yukteshwar";
  SiderealMode2[SiderealMode2["JNBhasin"] = 8] = "JNBhasin";
  SiderealMode2[SiderealMode2["BabylKugler1"] = 9] = "BabylKugler1";
  SiderealMode2[SiderealMode2["BabylKugler2"] = 10] = "BabylKugler2";
  SiderealMode2[SiderealMode2["BabylKugler3"] = 11] = "BabylKugler3";
  SiderealMode2[SiderealMode2["BabylHuber"] = 12] = "BabylHuber";
  SiderealMode2[SiderealMode2["BabylEtPSC"] = 13] = "BabylEtPSC";
  SiderealMode2[SiderealMode2["Aldebaran15Tau"] = 14] = "Aldebaran15Tau";
  SiderealMode2[SiderealMode2["Hipparchos"] = 15] = "Hipparchos";
  SiderealMode2[SiderealMode2["Sassanian"] = 16] = "Sassanian";
  SiderealMode2[SiderealMode2["GalacticCenter0Sag"] = 17] = "GalacticCenter0Sag";
  SiderealMode2[SiderealMode2["J2000"] = 18] = "J2000";
  SiderealMode2[SiderealMode2["J1900"] = 19] = "J1900";
  SiderealMode2[SiderealMode2["B1950"] = 20] = "B1950";
  SiderealMode2[SiderealMode2["SuryaSiddhanta"] = 21] = "SuryaSiddhanta";
  SiderealMode2[SiderealMode2["SuryaSiddhantaMeanSun"] = 22] = "SuryaSiddhantaMeanSun";
  SiderealMode2[SiderealMode2["Aryabhata"] = 23] = "Aryabhata";
  SiderealMode2[SiderealMode2["AryabhataMeanSun"] = 24] = "AryabhataMeanSun";
  SiderealMode2[SiderealMode2["SSRevati"] = 25] = "SSRevati";
  SiderealMode2[SiderealMode2["SSCitra"] = 26] = "SSCitra";
  SiderealMode2[SiderealMode2["TrueCitra"] = 27] = "TrueCitra";
  SiderealMode2[SiderealMode2["TrueRevati"] = 28] = "TrueRevati";
  SiderealMode2[SiderealMode2["TruePushya"] = 29] = "TruePushya";
  SiderealMode2[SiderealMode2["GalacticCenterGilBrand"] = 30] = "GalacticCenterGilBrand";
  SiderealMode2[SiderealMode2["GalacticEquatorIAU1958"] = 31] = "GalacticEquatorIAU1958";
  SiderealMode2[SiderealMode2["GalacticEquator"] = 32] = "GalacticEquator";
  SiderealMode2[SiderealMode2["GalacticEquatorMidMula"] = 33] = "GalacticEquatorMidMula";
  SiderealMode2[SiderealMode2["Skydram"] = 34] = "Skydram";
  SiderealMode2[SiderealMode2["TrueMula"] = 35] = "TrueMula";
  SiderealMode2[SiderealMode2["DhruvaGalCenterMulaWilhelm"] = 36] = "DhruvaGalCenterMulaWilhelm";
  SiderealMode2[SiderealMode2["Aryabhata522"] = 37] = "Aryabhata522";
  SiderealMode2[SiderealMode2["BabylBritton"] = 38] = "BabylBritton";
  SiderealMode2[SiderealMode2["UserDefined"] = 255] = "UserDefined";
  return SiderealMode2;
})(SiderealMode || {});
var RiseTransitFlag = /* @__PURE__ */ ((RiseTransitFlag2) => {
  RiseTransitFlag2[RiseTransitFlag2["Rise"] = 1] = "Rise";
  RiseTransitFlag2[RiseTransitFlag2["Set"] = 2] = "Set";
  RiseTransitFlag2[RiseTransitFlag2["UpperTransit"] = 4] = "UpperTransit";
  RiseTransitFlag2[RiseTransitFlag2["LowerTransit"] = 8] = "LowerTransit";
  return RiseTransitFlag2;
})(RiseTransitFlag || {});
var AsteroidOffset = 1e4;
var PlanetaryMoonOffset = 9e3;
var NumberOfPlanets = 23;

// src/implementations.ts
var LunarEclipseImpl = class {
  constructor(type, maximum, partialBegin, partialEnd, totalBegin, totalEnd, penumbralBegin, penumbralEnd) {
    this.type = type;
    this.maximum = maximum;
    this.partialBegin = partialBegin;
    this.partialEnd = partialEnd;
    this.totalBegin = totalBegin;
    this.totalEnd = totalEnd;
    this.penumbralBegin = penumbralBegin;
    this.penumbralEnd = penumbralEnd;
  }
  isTotal() {
    return (this.type & 4 /* Total */) !== 0;
  }
  isPartial() {
    return (this.type & 16 /* Partial */) !== 0;
  }
  isPenumbralOnly() {
    return (this.type & 64 /* Penumbral */) !== 0 && (this.type & (4 /* Total */ | 16 /* Partial */)) === 0;
  }
  getTotalityDuration() {
    if (!this.isTotal() || this.totalBegin === 0 || this.totalEnd === 0) {
      return 0;
    }
    const duration = (this.totalEnd - this.totalBegin) * 24;
    return duration > 0 ? duration : 0;
  }
  getPartialDuration() {
    if (this.partialBegin === 0 || this.partialEnd === 0) {
      return 0;
    }
    const duration = (this.partialEnd - this.partialBegin) * 24;
    return duration > 0 ? duration : 0;
  }
  getTotalDuration() {
    if (this.penumbralBegin === 0 || this.penumbralEnd === 0) {
      return 0;
    }
    const duration = (this.penumbralEnd - this.penumbralBegin) * 24;
    return duration > 0 ? duration : 0;
  }
};
var SolarEclipseImpl = class {
  constructor(type, maximum, partialBegin, partialEnd, centralBegin, centralEnd, centerLineBegin, centerLineEnd) {
    this.type = type;
    this.maximum = maximum;
    this.partialBegin = partialBegin;
    this.partialEnd = partialEnd;
    this.centralBegin = centralBegin;
    this.centralEnd = centralEnd;
    this.centerLineBegin = centerLineBegin;
    this.centerLineEnd = centerLineEnd;
  }
  isTotal() {
    return (this.type & 4 /* Total */) !== 0;
  }
  isAnnular() {
    return (this.type & 8 /* Annular */) !== 0;
  }
  isHybrid() {
    return (this.type & 32 /* AnnularTotal */) !== 0;
  }
  isPartial() {
    return (this.type & 16 /* Partial */) !== 0;
  }
  isCentral() {
    return (this.type & 1 /* Central */) !== 0;
  }
  isNonCentral() {
    return (this.type & 2 /* NonCentral */) !== 0;
  }
};
var DateTimeImpl = class {
  constructor(year, month, day, hour, calendarType = 1 /* Gregorian */) {
    this.year = year;
    this.month = month;
    this.day = day;
    this.hour = hour;
    this.calendarType = calendarType;
  }
  toISOString() {
    const hours = Math.floor(this.hour);
    const minutes = Math.floor((this.hour - hours) * 60);
    const seconds = Math.floor(((this.hour - hours) * 60 - minutes) * 60);
    const milliseconds = Math.floor((((this.hour - hours) * 60 - minutes) * 60 - seconds) * 1e3);
    const yearStr = Math.abs(this.year).toString().padStart(4, "0");
    const yearSign = this.year < 0 ? "-" : "";
    const monthStr = this.month.toString().padStart(2, "0");
    const dayStr = this.day.toString().padStart(2, "0");
    const hoursStr = hours.toString().padStart(2, "0");
    const minutesStr = minutes.toString().padStart(2, "0");
    const secondsStr = seconds.toString().padStart(2, "0");
    const msStr = milliseconds.toString().padStart(3, "0");
    return `${yearSign}${yearStr}-${monthStr}-${dayStr}T${hoursStr}:${minutesStr}:${secondsStr}.${msStr}Z`;
  }
  toString() {
    const calType = this.calendarType === 1 /* Gregorian */ ? "Gregorian" : "Julian";
    const yearStr = this.year < 0 ? `${Math.abs(this.year)} BCE` : this.year.toString();
    return `${yearStr}-${this.month.toString().padStart(2, "0")}-${this.day.toString().padStart(2, "0")} ${this.hour.toFixed(6)} hours (${calType})`;
  }
};

// src/flags.ts
var CalculationFlags = class _CalculationFlags {
  constructor(initialFlags) {
    this.flags = 0;
    if (initialFlags !== void 0) {
      this.add(initialFlags);
    }
  }
  /**
   * Add one or more flags to the current set
   * @param flag - Single flag or array of flags to add
   * @returns this (for method chaining)
   */
  add(flag) {
    if (Array.isArray(flag)) {
      flag.forEach((f) => this.flags |= f);
    } else {
      this.flags |= flag;
    }
    return this;
  }
  /**
   * Remove one or more flags from the current set
   * @param flag - Single flag or array of flags to remove
   * @returns this (for method chaining)
   */
  remove(flag) {
    if (Array.isArray(flag)) {
      flag.forEach((f) => this.flags &= ~f);
    } else {
      this.flags &= ~flag;
    }
    return this;
  }
  /**
   * Check if a specific flag is set
   * @param flag - Flag to check
   * @returns true if the flag is set
   */
  has(flag) {
    return (this.flags & flag) === flag;
  }
  /**
   * Convert to raw number for passing to C library
   * @returns The numeric representation of all combined flags
   */
  toNumber() {
    return this.flags;
  }
  /**
   * Create a new CalculationFlags instance from one or more flags
   * @param flags - Flags to combine
   * @returns New CalculationFlags instance
   */
  static from(...flags) {
    return new _CalculationFlags(flags);
  }
  /**
   * Common preset: Swiss Ephemeris with speed calculation
   */
  static get swissEphemerisWithSpeed() {
    return _CalculationFlags.from(2 /* SwissEphemeris */, 256 /* Speed */);
  }
  /**
   * Common preset: Moshier ephemeris with speed calculation
   */
  static get moshierWithSpeed() {
    return _CalculationFlags.from(4 /* MoshierEphemeris */, 256 /* Speed */);
  }
  /**
   * Common preset: Astrometric positions (no aberration or light deflection)
   */
  static get astrometric() {
    return _CalculationFlags.from(
      2 /* SwissEphemeris */,
      1024 /* NoAberration */,
      512 /* NoGravitationalDeflection */
    );
  }
  /**
   * Common preset: Heliocentric positions
   */
  static get heliocentric() {
    return _CalculationFlags.from(2 /* SwissEphemeris */, 8 /* Heliocentric */);
  }
  /**
   * Common preset: Topocentric positions
   */
  static get topocentric() {
    return _CalculationFlags.from(2 /* SwissEphemeris */, 32768 /* Topocentric */);
  }
  /**
   * Common preset: Equatorial coordinates (RA/Dec)
   */
  static get equatorial() {
    return _CalculationFlags.from(
      2 /* SwissEphemeris */,
      2048 /* Equatorial */,
      256 /* Speed */
    );
  }
};
var EclipseTypeFlags = class _EclipseTypeFlags {
  constructor(initialFlags) {
    this.flags = 0;
    if (initialFlags !== void 0) {
      this.add(initialFlags);
    }
  }
  /**
   * Add one or more eclipse types to the filter
   * @param flag - Single type or array of types to add
   * @returns this (for method chaining)
   */
  add(flag) {
    if (Array.isArray(flag)) {
      flag.forEach((f) => this.flags |= f);
    } else {
      this.flags |= flag;
    }
    return this;
  }
  /**
   * Check if a specific eclipse type is in the filter
   * @param flag - Eclipse type to check
   * @returns true if the type is included
   */
  has(flag) {
    return (this.flags & flag) === flag;
  }
  /**
   * Convert to raw number for passing to C library
   * @returns The numeric representation of all combined types
   */
  toNumber() {
    return this.flags;
  }
  /**
   * Create a new EclipseTypeFlags instance from one or more types
   * @param flags - Eclipse types to combine
   * @returns New EclipseTypeFlags instance
   */
  static from(...flags) {
    return new _EclipseTypeFlags(flags);
  }
  /**
   * Preset: All solar eclipse types
   */
  static get allSolar() {
    return new _EclipseTypeFlags([
      1 /* Central */,
      2 /* NonCentral */,
      4 /* Total */,
      8 /* Annular */,
      16 /* Partial */,
      32 /* AnnularTotal */
    ]);
  }
  /**
   * Preset: All lunar eclipse types
   */
  static get allLunar() {
    return new _EclipseTypeFlags([
      4 /* Total */,
      16 /* Partial */,
      64 /* Penumbral */
    ]);
  }
  /**
   * Preset: Only total eclipses
   */
  static get totalOnly() {
    return _EclipseTypeFlags.from(4 /* Total */);
  }
  /**
   * Preset: Total and partial eclipses (no penumbral)
   */
  static get totalAndPartial() {
    return _EclipseTypeFlags.from(4 /* Total */, 16 /* Partial */);
  }
};
function normalizeFlags(input) {
  if (typeof input === "number") {
    return input;
  }
  if (input instanceof CalculationFlags) {
    return input.toNumber();
  }
  if (Array.isArray(input)) {
    return CalculationFlags.from(...input).toNumber();
  }
  return input;
}
function normalizeEclipseTypes(input) {
  if (typeof input === "number") {
    return input;
  }
  if (input instanceof EclipseTypeFlags) {
    return input.toNumber();
  }
  if (Array.isArray(input)) {
    return EclipseTypeFlags.from(...input).toNumber();
  }
  return input;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Asteroid,
  AsteroidOffset,
  CalculationFlag,
  CalculationFlags,
  CalendarType,
  CommonCalculationFlags,
  CommonEclipseTypes,
  DateTimeImpl,
  EclipseType,
  EclipseTypeFlags,
  FictitiousPlanet,
  HousePoint,
  HouseSystem,
  LunarEclipseImpl,
  LunarPoint,
  NumberOfPlanets,
  Planet,
  PlanetaryMoonOffset,
  RiseTransitFlag,
  SiderealMode,
  SolarEclipseImpl,
  normalizeEclipseTypes,
  normalizeFlags
});
